roles/cluster_infra/templates/resources.tf.j2

```diff
@@ -233,6 +234,14 @@ resource "openstack_networking_port_v2" "{{ partition.name }}" {
 
 {% endfor %}
 
+#####
+##### Deploy key
+#####
+{% if cluster_ssh_private_key_file is not defined %}
```

Comment:
Should this be gated on whether `cluster_deploy_ssh_public_key` is given, rather than the private key file? Same where the private key is used.

Reply:
This needs to be gated on the private component of the key because of https://github.com/stackhpc/ansible-collection-terraform/blob/main/roles/infra/tasks/main.yml#L49.